vcd/resource_vcd_vapp_vm.go

```diff
@@ -498,128 +516,143 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
 	}
 
-	catalog, err := org.GetCatalogByName(d.Get("catalog_name").(string), false)
-	if err != nil {
-		return fmt.Errorf("error finding catalog %s: %s", d.Get("catalog_name").(string), err)
-	}
-
-	catalogItem, err := catalog.GetCatalogItemByName(d.Get("template_name").(string), false)
-	if err != nil {
-		return fmt.Errorf("error finding catalog item: %s", err)
-	}
+	catalogName := d.Get("catalog_name").(string)
+	templateName := d.Get("template_name").(string)
 
-	vappTemplate, err := catalogItem.GetVAppTemplate()
-	if err != nil {
-		return fmt.Errorf("error finding VAppTemplate: %s", err)
-	}
+	//create not empty VM - use provided template
+	if catalogName != "" && templateName != "" {
 
-	acceptEulas := d.Get("accept_all_eulas").(bool)
+		if catalogName == "" {
```

Comment:
The test for `empty catalogName` happens after we test that it has content (line 523).
Same for `templateName`

Reply:
I was misguided by another comment, yes you are right. Removing